tests/individual/TimeTravel.cpp

```diff
@@ -0,0 +1,145 @@
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+#include <nuclear>
+
+#include "test_util/TestBase.hpp"
+
+namespace {
+
+using TestUnits                = std::chrono::duration<int64_t, std::ratio<1, 50>>;
+constexpr int64_t EVENT_1_TIME = 4;
+constexpr int64_t EVENT_2_TIME = 8;
+
+struct Results {
+    struct TimePair {
+        NUClear::clock::time_point nuclear;
+        std::chrono::steady_clock::time_point steady;
+    };
+
+    TimePair start;
+    TimePair zero;
+    std::array<TimePair, 2> events;
+};
+
+class TestReactor : public test_util::TestBase<TestReactor, 5000> {
+public:
+    // Start time of steady clock
+    std::chrono::steady_clock::time_point steady_start_time = std::chrono::steady_clock::now();
+
+    // Time travel action
+    NUClear::message::TimeTravel::Action action = NUClear::message::TimeTravel::Action::RELATIVE;
+
+    // Time adjustment
+    NUClear::clock::duration adjustment = std::chrono::milliseconds(0);
+
+    // Real-time factor
+    double rtf = 1.0;
+
+    // Results struct
+    Results results;
+
+    TestReactor(std::unique_ptr<NUClear::Environment> environment) : TestBase(std::move(environment), false) {
+
+        on<Startup>().then([this] {
+            results.zero = Results::TimePair{NUClear::clock::now(), std::chrono::steady_clock::now()};
+
+            // Emit a chrono task to run at time EVENT_1_TIME
+            emit<Scope::DIRECT>(std::make_unique<NUClear::dsl::operation::ChronoTask>(
+                [this](NUClear::clock::time_point&) {
+                    results.events[0] = Results::TimePair{NUClear::clock::now(), std::chrono::steady_clock::now()};
+                    return false;
+                },
+                NUClear::clock::now() + TestUnits(EVENT_1_TIME),
```

Comment:
Since we're setting the time to 0 initially these can be set as absolute rather than relative event times
```suggestion
                NUClear::clock::time_point(TestUnits(EVENT_1_TIME)),
```